planner/core/rule_inject_extra_projection.go

```diff
@@ -61,9 +61,9 @@ func (pe *projInjector) inject(plan PhysicalPlan) PhysicalPlan {
 func injectProjBelowAgg(aggPlan PhysicalPlan, aggFuncs []*aggregation.AggFuncDesc, groupByItems []expression.Expression) PhysicalPlan {
 	hasScalarFunc := false
 
-	// If the mode is FinalMode, we do not need to wrap cast upon the args,
+	// If the mode is FinalMode or Partial2Mode, we do not need to wrap cast upon the args,
 	// since the types of the args are already the expected.
-	if len(aggFuncs) > 0 && aggFuncs[0].Mode != aggregation.FinalMode {
+	if len(aggFuncs) > 0 && aggFuncs[0].Mode != aggregation.FinalMode && aggFuncs[0].Mode != aggregation.Partial2Mode {
```

Comment:
We don't need projection for `DedupMode` either?

Reply:
DedupMode has not been used now.